### PR TITLE
Recalc height after hidden init

### PR DESCRIPTION
**Pull request by emil-furniss**
Picking up from #26, @mdulat found that the issue was not the loading with empty content after all, but that the fix is still required. Te problem happens when a parent element is `hidden`, which causes `offsetHeight` to be set to 0 for all decedents.
A unit test is in place to test that, along with the original fix of recalculating `__baseHeight` when an adjustment is made and it was originally set to 0.

### CODE REVIEW

File: d2l-more-less.html [hunk 1]
```diff
@@ -287,6 +287,12 @@
 				return;
 			}
 
+			if (this.__baseHeight === 0) {
```
Comment (dlockhart) [inline d2l-more-less.html]: When you're doing mutations or reads to things that cause layout recalculations/repaints, you should wrap them in `fastdom.mutate` and `fastdom.measure`. That way the mutates get batched together and the reads do as well, causing minimum jank. There are a few examples of it in use in more/less already.

Comment (dlockhart) [reply d2l-more-less.html]: In this case:

```
fastdom.mutate(function() {
    this.__content.style.height = this.height;
    fastdom.measure(function() {
        this.__baseHeight = this.__content.offsetHeight;
        fastdom.mutate(function() {
            this.__content.style.height = null;
        }.bind(this);
    }.bind(this));
}.bind(this));

Comment (omsmith) [reply d2l-more-less.html]: Sounds like we just want...
```js
fastdom.mutate(this.__init_setBaseHeight, this);
return;
```

Which then goes to say (and judging by the name of the PR), do we just need to defer when that gets run initially?

File: d2l-more-less.html [hunk 2]
```diff
@@ -130,7 +130,9 @@
 			this.__content = this.$$('.more-less-content');
 			this.__contentSlot = this.$$('.more-less-content slot');
 
-			fastdom.mutate(this.__init_setBaseHeight, this);
+			if (this.__content.offsetParent === null) {
```
Comment (omsmith) [inline d2l-more-less.html]: Was this meant to be `!== null`?

Comment (emil-furniss) [reply d2l-more-less.html]: of course it was, my bad.

File: test/d2l-more-less.html [hunk 3]
```diff
@@ -60,6 +69,23 @@
...
 				});
+				describe('initially-hidden', function() {
+                    beforeEach(function() {
+                        div = fixture('initially-hidden');
+                        elem = div.getElementsByTagName('d2l-more-less')[0];
+                    });
+                    it('should be inactive when content is short and initially hidden', function(done) {
+                        setTimeout(function () {
+                            expect(elem.__baseHeight).to.equal(0);
+                            div.style.display = "block";
```
Comment (omsmith) [inline test/d2l-more-less.html]: What observer is picking up this change?

Comment (emil-furniss) [reply test/d2l-more-less.html]: ResizeObserver

Comment (dlockhart) [reply test/d2l-more-less.html]: Fancy!

**Pull request merged**